### PR TITLE
Support multiple peer experiences powered by a single server

### PR DESCRIPTION
**Pull request by bengreenier**
> Note: cutting this PR so we can track this merge and store context about work here 🚀 

Goals:

- [x] update server framework code to support multi peer
- [x] update server samples to support multi peer
- [x] update signalling code to support multi peer
- [ ] update feature matrices
- [x] document featureset/backcompat changes
    - documented basics of multi-peer configuration @ https://github.com/CatalystCode/3dtoolkit/wiki/Multi-peer

### CODE REVIEW

**Comment by phongcao** [comment]
@bengreenier Could you help review my update?
@tylergibson @anderm Could you guys help with code review and testing? I haven't tested the case with multiple HoloLens but it should work.

**Comment by tylergibson** [comment]
@phongcao you've got a build break with the CI build :)

**Comment by phongcao** [comment]
@tylergibson Yeah, I'm working on it now. Hopefully everything will be ready for you and Andrei to test today.

**Comment by tylergibson** [comment]
Still seeing server crash after first session.  If a client connects (tested with Win32 DirectX client) and then disconnects, and reconnects to same server, server will crash.  Affects every server sample.

Also, even running the multi-peer signaling server, clients are never cleaned out. After each session, an additional duplicate client and server entry are added.  It looks like we're missing the hangup/disconnect  logic for the signaling server.

**Comment by phongcao** [comment]
@tylergibson Not sure I can reproduce the server crash due to re-connecting as you mention. What kind of exception do you see when it occurs? Agree that there is a need to do the cleanup properly for hangup/disconnect.

**Comment by phongcao** [comment]
@anderm @tylergibson Please help review the code and do a lot of testing! :)

**Comment by bengreenier** [comment]
@tylergibson sorry for the delay wrt signalling stuff, i fractured a bone and had to take some time off 🍖 

>  It looks like we're missing the hangup/disconnect logic for the signaling server.

you're correct sir. the quick workaround during the hack was to remove the cleanup logic making it easy to connect to the same server over and over. 

The official ™️ solution is going to leverage https://github.com/bengreenier/webrtc-signal-http-capacity and is being tested by myself. I'm not quite ready to sign off as 100% stable yet, should be soon. 💯

**Comment by phongcao** [comment]
@tylergibson I also need more time to debug the connection issues what you found so let's target to have everything merged this week.

**Comment by tylergibson** [comment]
@bengreenier Need to add the capacity feature to the default serverconfig.json across all server plugins so it has a default and can be easily altered at runtime.  Also, the hololens client is no longer functional - there is no way for it to connect to a MP server as the server UI cannot be used to connect to clients now?  

Autocall (in flagdefs.h) seems to be non-functional with the new MP code, again preventing UI-less clients like DXHoloLens from connecting.

@phongcao 
If you try to connect two DXWin32 clients to the same Server (multhreaded, all win64) on the same machine, the second client will crash if you disconnect the session and try to reconnect with it.

**Comment by bengreenier** [comment]
Updated https://github.com/CatalystCode/3dtoolkit/wiki/Multi-peer and https://github.com/CatalystCode/3dtoolkit/wiki/JSON-Config-Files#server-configuration-serverconfigjson for capacity configuration information

**Comment by bengreenier** [comment]
@tylergibson moved autoconnect and autocall to config, as we discussed offline. it actually made alot of this feel cleaner, imo.

Updated https://github.com/CatalystCode/3dtoolkit/wiki/JSON-Config-Files#server-configuration-serverconfigjson

File: Plugins/NativeServerPlugin/src/directx_multi_peer_conductor.cpp [hunk 1]
```diff
@@ -195,14 +201,10 @@ void DirectXMultiPeerConductor::DisconnectFromServer() {}
 
 void DirectXMultiPeerConductor::ConnectToPeer(int peer_id)
 {
-	auto it = connected_peers_.find(peer_id);
-	if (it != connected_peers_.end())
+	auto peer = SafeAllocatePeerMapEntry(peer_id);
```
Comment (bengreenier) [inline Plugins/NativeServerPlugin/src/directx_multi_peer_conductor.cpp]: the implementation that was here would always fail, as when you are connecting to a client there is not yet a message from the client, therefore there is no entry here yet. by using the new SafeAllocatePeerMapEntry we create an entry if needed, or use the existing if found

**Comment by phongcao** [comment]
Thanks Ben! Let's try to get this one merged to master soon.

**Comment by bengreenier** [comment]
@phongcao what's the current status here? Do you think we are ready to merge?

**Comment by phongcao** [comment]
I'm still working on the Unity FractalStream server. After I've finished it and maybe @tylergibson approves, it'll be good to go.

**Comment by phongcao** [comment]
I'm quite happy with the Unity FractalStream + DirectX HoloLens client performance now so I guess the last issues for us are the ones that @tylergibson found related to VM.

**Pull request merged**